include/dylib.hpp

```diff
@@ -227,13 +227,12 @@ class dylib {
         FreeLibrary(lib);
     }
     static char *_get_error() noexcept {
-        constexpr size_t buf_size = 512;
         auto error_code = GetLastError();
         if (!error_code)
             return nullptr;
-        static char msg[buf_size];
+        char msg[512];
```

Comment:
This is a bug. You are now returning a pointer to a local variable which goes out of scope.